lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadata.php

```diff
@@ -329,6 +329,13 @@ class ClassMetadata implements ClassMetadataInterface
     public $parentClasses = array();
 
     /**
+     * READ-ONLY: Child class restrictions.
+     *
+     * @var array
+     */
+    public $childRestrictions = null;
```

Comment:
actually yes, lets only use whitelisting then. finer grained stuff can
be done with listeners or custom validators and whatnot. and go with
requiredClasses - i like to keep this as close to PHPCR as makes sense.

but i think PHPCR also does the equivalent of an instanceof check, so we
want to keep doing that.